src/main/drivers/serial_uart_impl.h

```diff
@@ -17,10 +17,154 @@
 
 #pragma once
 
-// device specific uart implementation is defined here
+// Configuration constants
+
+#if defined(STM32F1)
+#define UARTDEV_COUNT_MAX 3
+#define UARTHARDWARE_MAX_PINS 3
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     256
+#endif
+#ifndef UART_TX_BUFFER_SIZE
+#define UART_TX_BUFFER_SIZE     256
+#endif
+#elif defined(STM32F3)
+#define UARTDEV_COUNT_MAX 5
+#define UARTHARDWARE_MAX_PINS 4
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     256
+#endif
+#ifndef UART_TX_BUFFER_SIZE
+#define UART_TX_BUFFER_SIZE     256
+#endif
+#elif defined(STM32F4)
+#define UARTDEV_COUNT_MAX 6
+#define UARTHARDWARE_MAX_PINS 4
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     512
```

Comment:
do sizes over 256 work now? See comments at the top of serial_uart.h regading using uint8_t as parameters and return types from various serial IO routines. If so then delete/update the comments otherwise set the ones > 256 to 256.